scripts/check-test-complexity.sh

```
@@ -0,0 +1,99 @@
+
+# Check 5: No complex abstractions in test files (exclude e2e/helpers directory)
+ABSTRACT_COUNT=$(grep -r "class.*Test\|extends.*Test\|abstract.*test" tests/*.test.js 2>/dev/null | wc -l || echo "0")
+if [ "$ABSTRACT_COUNT" -gt 0 ]; then
+  echo "❌ Found test abstractions/classes ($ABSTRACT_COUNT occurrences)"
+  COMPLEXITY_ISSUES=$((COMPLEXITY_ISSUES + 1))
+else
+  echo "✅ No complex test abstractions found"
+fi
+
+# Check 6: No complex mocking frameworks (exclude e2e/helpers directory)
+MOCK_COUNT=$(find tests -name "*.test.js" -o -name "setup.js" -o -name "helpers.js" -o -name "vitest.config.js" | xargs grep -h "jest\.mock\|sinon\|proxyquire\|mockery" 2>/dev/null | wc -l || echo "0")
```

Shellcheck (0.10.0)

[warning] Use 'find .. -print0 | xargs -0 ..' or 'find .. -exec .. +' to allow non-alphanumeric filenames.

(SC2038)